### PR TITLE
Fix modal usage and remove Cypress workaround

### PR DESCRIPTION
**Pull request by andresriveratoro**
## Description
This PR updates component-library to the latest version, removes the role attribute from the modal and removes Cypress workaround implemented in https://github.com/department-of-veterans-affairs/vets-website/pull/21925. This attribute should not be set manually and using it results in accessibility tests failing.

## Original issue(s)
https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1113

## Screenshots
Screen Shot 2022-08-23 at 17 34 03

## Acceptance criteria
- [x] component-library passes Cypress tests

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/forms-system/src/js/utilities/ui/index.js [hunk 1]
```diff
@@ -115,7 +115,22 @@ export function scrollToFirstError() {
...
+        'va-modal[visible]:not([visible="false"])',
+      ),
+    );
+
+    const isModalOpen =
+      document.body.classList.contains('modal-open') ||
+      document.querySelectorAll('va-modal[visible]:not([visible="false"])') ||
+      isShadowRootModalOpen;
+
+    if (!isModalOpen) {
```
Comment (bkjohnson) [inline src/platform/forms-system/src/js/utilities/ui/index.js:133]: Have you tested that the `isModalOpen` condition will be truthy if a `<va-modal>` is open and the `<body>` doesn't have the `modal-open` class?

Comment (andresriveratoro) [reply src/platform/forms-system/src/js/utilities/ui/index.js:133]: @bkjohnson I've updated the query selectors and tested that they work as expected on vets-website. I did not test them in the context of the `scrollToFirstError` function.

Comment (bkjohnson) [reply src/platform/forms-system/src/js/utilities/ui/index.js:133]: I didn't run the full function but I did a quick test with a `<va-modal>` on a page and calculated `isModalOpen` in the console and it gave us `true` like we wanted.

```diff
diff --git a/src/applications/coronavirus-vaccination-expansion/containers/IntroductionPage.jsx b/src/applications/coronavirus-vaccination-expansion/containers/IntroductionPage.jsx
index 365c6650fb..5ff80a3c21 100644
--- a/src/applications/coronavirus-vaccination-expansion/containers/IntroductionPage.jsx
+++ b/src/applications/coronavirus-vaccination-expansion/containers/IntroductionPage.jsx
@@ -66,6 +66,7 @@ class IntroductionPage extends React.Component {
   render() {
     return (
       <div className="schemaform-intro">
+        <va-modal visible>Testing</va-modal>
         <br />
         <FormTitle title="Sign up to get a COVID-19 vaccine at VA" />
         <span>
```


![Screenshot of a local page with a modal open and the console open in the browser dev tools. The JS used to determine if a modal was open which uses a `va-modal[visible]` query selector is pasted into the console and the output shows that the boolean is true.](https://user-images.githubusercontent.com/2008881/187306021-8c9ac8ba-b6d0-4dcc-936e-c9d17a4fbab0.png)

I'll approve the other PR so we can make a release and bump this version again.

**Review by bkjohnson — APPROVED** [review]

**Review by bjmccotter7192 — APPROVED** [review]

**Review by alexandec — APPROVED** [review]

**Review by vbahinwillit — APPROVED** [review]
LGTM for VAOS changes

**Review by k80bowman — APPROVED** [review]

**Pull request merged**